### PR TITLE
[WIP] Psych Hospital/Medical Lab Dungeon

### PR DESCRIPTION
**Pull request by worm-girl**
#### Summary
Content "Adds a globally unique new XEDRA lab"

#### Purpose of change

TCL made it clear that our modern EOCs and mapping tools make it possible to create content that blows the semi-obsolete old labs out of the water. During a discussion on mutagen loot, it was pointed out that unlike Alpha (one of the other "advanced" lines) there's no reliable place to get Medical mutagen.

#### Describe the solution

THE PSYCHIATRIC HOSPITAL
This PR will add a new globally unique location based very loosely on the historic [Danvers State Hospital](https://www.danversstatehospital.org/), a famous psychiatric hospital in Massachusetts. The real-life Danvers was sadly demolished in 2006, but it can serve as a helpful blueprint here. If you're unfamiliar, the hospital was a four-story brick building on a large campus with a network of concrete tunnels beneath it. It served as the setting for the 2001 film Session 9, which would go on to be a big visual inspiration for Silent Hill 3. Our hospital isn't Danvers, but that's about the flavor we're going for. 

The hospital is a two story brick building roughly the size of a mansion, but unlike a mansion, the inside is fairly claustrophobic, designed to confuse patients about their orientation and prevent escape. This is a real thing and I forget what it's called, but if you look at the map, it is tactically pretty different from our existing mansions.

The upper levels are almost completely inhabited by regenerating zombies, with some amalgamations and other Zed-aligned nasties. There are also a few regenerating mutants - humans who took so much medical mutagen that they've gone permanently bonkers, like the other lab mutants you can find in TCL. There are also two mutated sharpshooters on the roof, former security guards who are still following orders to prevent unauthorized escape or break-ins. These will be removed if they don't play nice, my hope is that they create area denial so the player has to approach at night or by being clever about cover, unless they just want to armor up and sprint for the entrance.

THE LAB
There's a three-level lab beneath the hospital. B1 is mostly just a security checkpoint, B2 holds subject cells, a chemistry lab, a pathology lab, a crematorium, staff housing, offices, and some other stuff. B3 is an anteroom and the mutagen vault, where Dr. Isha is hiding. He has mutated way too much and cannot be reasoned with. He talks to the player if he sees them on CCTV (something I'll do via EOCs) as they explore the upper levels, giving the player ample warning that he has a fairly powerful gun and will shoot at them if they unlock his door.

Isha himself does good damage at medium range, can handle himself OK up close, and most importantly, he regenerates very quickly. As a threat, he's weaker than the melded task force and lacks their ubergrab ability, but guns are always a wildcard. Additionally, he's inside a room that is filled with glass containers of the mutagen you're probably hoping to grab. You can just kill him with a grenade or rocket, but then you lose a lot of good loot. Currently he regenerates fast enough that you either need to be a crack shot with a machine gun or you need a partner to take him down. Since he's confined to an area he can't escape from without the player letting him free, I think it's fair to ask that of them.

In any event, players might enjoy the chance to punish an actual living XEDRA scientist for his crimes against humanity, even if he's William Birkin'd himself.

This also adds a new trap, tr_weak_floor, which uses the magic system (all json baby!!). If you plus your gear weigh over 86182 grams, the floor will collapse and you'll fall through to the level below.

#### Describe alternatives you've considered

None so far. I am only about halfway done with the mapping, I still need to do itemgroups and finish the lower two levels before I can even get started on the EOCs and snippets, but there's enough here to glance through if you're curious. I'm posting this earlier than I otherwise would as I'd rather know now if anything needs to be changed, or if anyone has anything to contribute.

The story is fairly pulpy, maybe a little too pulpy, but this is a horror game. In any event it will be clear that XEDRA did not set out to create an evil cenobite factory, the place went from a sterile sort of everyday military industrial complex evil to grindhouse chaos only after blob psychosis started to take root. I will also stringently avoid insensitive depiction of the patients' prior afflictions - they are normal people who were victimized by the system, not monsters. Or they were anyway, now everyone is monsters.

#### Testing

Ongoing

#### Additional context

LORE BACKGROUND
>  "The research team headed by Dr. Isha has produced a concentrated form of mutagen that shows promising results for the treatment of many forms of disease.  Testing reveals that it inhibits the body's pain responses while bolstering the immune system and natural regenerative capabilities, though a few troubling behavioral side effects have been reported."

When buying up land to conduct their research, XEDRA selected the site of a century-old psychiatric hospital that was, like many others, closed during the Reagan administration. A fake nonprofit was cooked up, and following renovations, reopened the site as a modern day place of healing for those with severe mental disorders. The fact that they were shouldering the burden of housing and caring for some of the most disadvantaged people in the country gave them enormous latitude to quietly disappear some of their patients. The project was headed by Dr. Isha, the canonical inventor of Medical mutagen, who served double duty as both the head of a lab team and the director of the hospital.

The upper two levels were run by real mental health professionals who actually did their jobs. Patients (most of them foisted off on the hospital by cities eager to reduce the number of visible homeless) received quality care. Some of them even got better and were allowed to leave. The basement, however, contains a three-level XEDRA lab. A small number of patients who were deemed untreatable would be sent below for "experimental treatment."

HOW IT'S MADE
Mutagen in the lore is generally 'typed' by exposing the raw stuff to animal or plant tissue under specific circumstances. We've never really established how medical mutagen works, except that drugs are involved. I don't intend to get too deep into the process, but I want to leave some snippets and environmental storytelling around that suggest the process involved taking a sick or injured person, administering drugs to treat them, then exposing their tissue samples to raw mutagen. This creates a mutagen which can cause the body to effectively treat itself.

STORY SO FAR
Isha believed his team had developed the ultimate medicine. No matter what they threw at their test subjects, the medicine cured everything. There were still some unpleasant side effects to work out, but that would all come in time. And then he got diagnosed with glioblastoma.

Unexpectedly given just a few months to live, Isha did what any good science fiction villain would do and dosed himself with his own experimental superdrug. This worked, but caused him to rapidly lose his marbles. Order broke down in the hospital as he started dosing patients and staff indiscriminantly, building up sort of a cult of personality as he slowly turned into basically Pinhead from Hellraiser, terrorizing his subjects (many of whom were fully on board once they became psychotic regenerating pain junkies) with everything from vivisection to lobotomy as he tried to test the limits of this miracle drug.

His superiors caught wind of this pretty fast - no XE037 was ever supposed to leave the lower levels for fear of contamination - and sent a killteam (think Hunk from Resident Evil) to take him out, but 5.7mm doesn't work on a guy who has gone well beyond the Medical threshold and has a bunch of regenerating security officers to back him up. He managed to kill them, but this was a week before the Cataclysm. Most of the hospital's residents went feral and started tearing each other to pieces. Those lucky enough to die were soon reborn as amalgamations or regenerating zombies. Isha retreated to the lowest levels of the lab, locked himself in the mutagen vault, and now spends his days eating canned food and watching over his little kingdom on CCTV.

https://www.youtube.com/watch?v=nTVTtakCgmM

### CODE REVIEW

**Comment by Zireael07** [comment]
Looks like a very flavorful place. I <3 <3 <3

File: data/json/monsters/mutant.json [hunk 1]
```diff
@@ -324,5 +324,90 @@
     "delete": { "flags": [ "HIT_AND_RUN" ], "fear_triggers": [ "FIRE" ] },
     "extend": { "flags": [ "SMALLSLUDGETRAIL", "SLUDGEPROOF", "NOHEAD", "STUN_IMMUNE", "FIREPROOF" ] },
     "armor": { "bash": 22, "cut": 45, "stab": 30, "heat": 50, "bullet": 70, "electric": 6 }
+  },
+  {
+    "id": "mon_mutant_isha",
+    "type": "MONSTER",
+    "name": { "str": "Dr. Isha" },
```
Comment (Qrox) [inline data/json/monsters/mutant.json:331]:
```suggestion
    "name": { "str": "Dr. Isha", "str_pl": "Drs. Isha" },
```

**Comment by Maleclypse** [comment]
`historic [Danvers State Hospital](https://www.danversstatehospital.org/), a famous psychiatric` Ha this was originally where I was going to make the ratkin lab but I made it a generic microlab instead to get it out faster.  Great choice!

File: data/json/overmap/overmap_terrain/overmap_terrain_public_institutional.json [hunk 2]
```diff
@@ -695,5 +695,62 @@
...
+      "lab_psych_hospital_4f_1",
+      "lab_psych_hospital_4f_2",
+      "lab_psych_hospital_4f_3"
+    ],
+    "name": "psychiatric hospital",
+    "sym": "H",
+    "color": "blue"
   }
 ]
+]
```
Comment (Maleclypse) [inline data/json/overmap/overmap_terrain/overmap_terrain_public_institutional.json:756]:
```suggestion
]

```

File: data/json/overmap/overmap_special/specials.json [hunk 3]
```diff
@@ -7262,5 +7262,63 @@
...
+      { "point": [ 0, 1, -2 ], "overmap": "lab_psych_hospital_b2_6_north" },
+      { "point": [ 2, 2, -2 ], "overmap": "lab_psych_hospital_b2_7_north" },
+      { "point": [ 1, 2, -2 ], "overmap": "lab_psych_hospital_b2_8_north" },
+      { "point": [ 0, 2, -2 ], "overmap": "lab_psych_hospital_b2_9_north" }
+    ],
+    "occurrences": [ 75, 100 ],
+    "flags": [ "CLASSIC", "MAN_MADE", "GLOBALLY_UNIQUE" ]
   }
 ]
+]
```
Comment (Maleclypse) [inline data/json/overmap/overmap_special/specials.json:7324]:
```suggestion
]

```

**Comment by fairyarmadillo** [comment]
> Awww what happen?

Four of my merged PRs were reverted in two days and there's an IMO very inadvisable PR aimed specifically at me trying to impose a hard limit of 500 lines on PRs for all contributors. I can't help but feel that my contributions aren't welcome here, and at the very least I can't find the motivation to continue at this time.

@Maleclypse or someone else may want to pick this up at some point since they were thinking of making a place like this. If so, the current state of the PR is that the upper levels are basically done - the west side of 2f needs some attention - it's supposed to be a disused part of the hospital with a bunch of gurneys crammed in everywhere blocking off rooms and hallways almost like a sokoban puzzle, but otherwise the rest of the place is about like I envisioned it. The main thing left to do would be to change the item spawns and start placing monsters (the idea was to put a lot of regenerating zombies), graffiti and notes about the hospital full of medical mutants rapidly losing their minds, and computer log entries.

The basement currently only holds the lab entrance. I made another lab floor on my local copy, but I've lost the repository this was being done on and I can't find the motivation to fix all of that. The intent was to have a fairly small lab with medical/surgical themeing, and then on the floor below that would be the mutagen vault with Dr. Isha. He was going to watch the player on CCTV and say creepy stuff over the intercom via EOCs as you explored and made your way down to him.

Maybe someone will make something happen, maybe not. But the work is here if anyone wants to use it, whether for this purpose or any other.

**Comment by Maleclypse** [comment]
> >  IMO very inadvisable PR aimed specifically at me trying to impose a hard limit of 500 lines on PRs for all contributors. I can't help but feel that my contributions aren't welcome here, and at the very least I can't find the motivation to continue at this time.

I want you to know that the guideline "not a hard limit" is not aimed at you at all.  It's a request you can find littered throughout the github and on the discord.  The PR adding it to the PR template is for the following long winded reason.  

Fairyarmadillo, 
It is no one's intention to make you feel bad, and yet this has clearly made you feel bad.  No one likes making someone feel bad about contributing and that makes many of us feel bad when that occurs.  Our hope is that in adding language to the template suggesting smaller PRs that we will make less people feel unwanted here by attempting to keep things in sizes that are easier to review.  I want to apologize to you because I was excited for the concepts you brought and where I was unsure about the c++ or content I relied upon just compiling and testing it, which in several cases was not enough.  I am sorry for that.  You feel how you feel and your feelings are valid and I apologize for my part in those feelings.  I and many others appreciate your contributions and I hope that one day you feel good about coming back and that we as a community are better able to work together with you in that future.  Good luck, I'm sorry and thank you.

**Comment by fairyarmadillo** [comment]
Thanks, accepted, and I appreciated working together. It is more the reversions than anything else - the seed one was totally justified given the situation, but seeing three others come down in a 24 hour period, and _only_ for my work, including stuff that's been merged for months and generally pretty well-received, including some mischaracterization in the PR bodies - I can't help taking that personally, especially when I'm just getting stonewallled in the comments. A single reversion, or better yet an invitation to resubmit my work with some guidelines for what ought to be improved, would have been fine. Four at once, only involving my stuff, and with the line limit PR as the cherry on top, is just deflating.

**Comment by KittyTac** [comment]
> Thanks, accepted, and I enjoyed working together. It is more the reversions than anything else. The seed one was totally justified given the situation, but seeing three others come down in a 24 hour period, and _only_ for my work, including stuff that's been merged for months and generally pretty well-received, including some mischaracterization in the PR bodies, well, I can't help taking that a certain way, especially when I'm just getting stonewalled in the comments. A single reversion, or better yet an invitation to resubmit my work with some guidelines for what ought to be improved, would have been fine. Four at once, only involving my stuff, and with the line limit PR as the cherry on top even after I'd split all my PRs is just deflating.
> 
> I have done a lot more work than is in those reversion PRs. I had the armor changes in #72010 that corrected some longstanding bugs and set up more logical behavior for acid. I had coded the ability for fields to cast spells like traps do, which would allow them to do all kinds of things via json (and unhardcode acid completely). I made EOCs and iexamines for washing bile and acid off. I had a long list of changes prepped for survivor gear aimed at making it homemade soft armor that could reasonably deal with acid. But that was all tied to the splash stuff which didn't seem particularly controversial in either PR, and any issues with it seem like they could have been addressed via issues rather than just throwing the baby out with the bathwater.
> 
> None of that's on you IMO, it just sucks generally.

Yeah, I empathize.

I do hope you return eventually, I really liked your work. Stay safe.

**Pull request closed without merging**